backend/source/archive/models/google_takeout.py

```diff
@@ -179,31 +195,34 @@ def extract_history_entries(self, json_files: Iterable[Path], schema: str,
                                 prefix: str) -> Generator[Entry, None, None]:
         for json_file in json_files:
             logger.info(f'Processing activity in "{str(json_file)}"')
-            for entry in json.load(json_file.open('r')):
-                if entry['title'].startswith(prefix):
-                    try:
-                        time = pytz.utc.localize(datetime.strptime(entry['time'], '%Y-%m-%dT%H:%M:%S.%fZ'))
-                    except ValueError:
-                        time = json_to_datetime(entry['time'])
-
-                    extra_attributes = {}
-                    if entry.get('titleUrl'):
-                        extra_attributes['url'] = entry['titleUrl']
-
-                    try:
-                        yield Entry(
-                            title=entry['title'].replace(prefix, '', 1),
-                            description='',
-                            source=self.entry_source,
-                            schema=schema,
-                            date_on_timeline=time,
-                            extra_attributes=extra_attributes
-                        )
-                    except KeyboardInterrupt:
-                        raise
-                    except:
-                        logging.exception(f"Could not parse entry: {entry}")
-                        raise
+            try:
+                for entry in json.load(json_file.open('r')):
+                    if entry['title'].startswith(prefix):
+                        try:
+                            time = pytz.utc.localize(datetime.strptime(entry['time'], '%Y-%m-%dT%H:%M:%S.%fZ'))
+                        except ValueError:
+                            time = json_to_datetime(entry['time'])
+
+                        extra_attributes = {}
+                        if entry.get('titleUrl'):
+                            extra_attributes['url'] = entry['titleUrl']
+
+                        try:
+                            yield Entry(
+                                title=entry['title'].replace(prefix, '', 1),
+                                description='',
+                                source=self.entry_source,
+                                schema=schema,
+                                date_on_timeline=time,
+                                extra_attributes=extra_attributes
+                            )
+                        except KeyboardInterrupt:
+                            raise
+                        except:
+                            logging.exception(f"Could not parse entry: {entry}")
+                            raise
+            except:
```

Comment:
Don't wrap huge chunk of code in a catch-all exception handler. It will silence unexpected bugs.

http://oncodingstyle.blogspot.com/2008/10/fail-early-fail-loudly.html